Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/PressableButtonInspector.cs

```diff
@@ -61,120 +97,135 @@ private void OnSceneGUI()
+            info.touchStartPos = info.touchStartOrigin.z;
+            info.releaseDistPos = info.pressDistPos - (info.releaseDistanceDelta / transform.lossyScale.z);
+
+            return info;
+        }
+
+        private void DrawButtonInfo(ButtonInfo info, bool editingEnabled)
+        {
+            // This is where we'll store our new values to compare against
+            ButtonInfo newInfo = info;
 
             if (editingEnabled)
             {
                 EditorGUI.BeginChangeCheck();
             }
-
-            float handleSize = touchCageSize.x * 0.065f;
-
-            // END POINT
+            
+            // PRESS END
             Handles.color = Color.cyan;
-            DrawPlaneAndHandle(targetEndPlane, touchCageSize * 0.5f, handleSize, ref newEndPos, "Max move distance");
+            DrawPlaneAndHandle(targetEndPlane, info.touchCageLocalBounds.size * 0.5f, ref newInfo.endPos, info.touchStartOrigin, "Max move distance", editingEnabled);
 
             if (editingEnabled)
             {
                 // Clamp the z value to start position
-                newEndPos.z = Mathf.Max(startPos.z, newEndPos.z);
-                newMaxPushDistance = Vector3.Distance(startPos, newEndPos);
+                newInfo.endPos = Mathf.Max(newInfo.startPos, newInfo.endPos);
+                newInfo.maxPushDistance = Mathf.Abs(newInfo.startPos - newInfo.endPos) * transform.lossyScale.z;
             }
 
             // PRESS DISTANCE
             Handles.color = Color.yellow;
-            DrawPlaneAndHandle(pressDistancePlane, touchCageLocalBounds.size * 0.35f, handleSize, ref newPressDistPos, "Press event");
+            DrawPlaneAndHandle(pressDistancePlane, info.touchCageLocalBounds.size * 0.35f, ref newInfo.pressDistPos, info.touchStartOrigin, "Press event", editingEnabled);
 
             if (editingEnabled)
             {
                 // Clamp the z values to target start / end
-                newPressDistPos.z = Mathf.Max(startPos.z, newPressDistPos.z);
-                newPressDistPos.z = Mathf.Min(newEndPos.z, newPressDistPos.z);
+                newInfo.pressDistPos = Mathf.Max(newInfo.startPos, newInfo.pressDistPos);
+                newInfo.pressDistPos = Mathf.Min(newInfo.endPos, newInfo.pressDistPos);
                 // Set based on distance from start
                 // Adjust for scaled objects
-                newPressDistance = Vector3.Distance(newPressDistPos, startPos) * transform.lossyScale.z;
+                newInfo.pressDistance = Mathf.Abs(newInfo.pressDistPos - newInfo.startPos) * transform.lossyScale.z;
             }
 
             // RELEASE DISTANCE DELTA
             Handles.color = Color.red;
-            DrawPlaneAndHandle(releasePlane, touchCageLocalBounds.size * 0.3f, handleSize, ref newReleaseDistPos, "Release event");
+            DrawPlaneAndHandle(releasePlane, info.touchCageLocalBounds.size * 0.3f, ref newInfo.releaseDistPos, info.touchStartOrigin, "Release event", editingEnabled);
```

Comment:
what is this 0.3f representing? Could we maybe take this magic number and have it be a more descriptive variable?

Reply:
It just represents the size of the plane.

Using magic numbers to adjust SceneGUI handles is another case where I'm following conventions set in other inspectors. No problem with ripping out magic numbers but if we want to do so then we should open a separate PR, establish a style guide and homogenize all our existing inspectors.